packages/memory/src/index.ts

```diff
@@ -139,16 +140,25 @@ export class Memory extends MastraMemory {
       threadConfig: config,
     });
 
-    // First sort messages by date
     const orderedByDate = rawMessages.sort((a, b) => new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime());
-    // Then reorder tool calls to be directly before their results
-    const reorderedToolCalls = reorderToolCallsAndResults(orderedByDate);
 
-    // Parse and convert messages
-    const messages = this.parseMessages(reorderedToolCalls);
-    const uiMessages = this.convertToUIMessages(reorderedToolCalls);
-
-    return { messages, uiMessages };
+    const list = new MessageList({ threadId, resourceId }).add(orderedByDate, 'memory');
+    return {
+      get messages() {
+        const v1Messages = list.get.all.v1();
```

Comment:
why do we return v1 and not v2 messages here?

Reply:
For backwards compatibility - in a breaking change we can change it to always surface ui messages instead, probably we can do that after SDK V5

Reply:
There are folks who are calling this method as part of our public API

Reply:
I'll add a better comment, thanks for the q!